provider/clinia/internal/codec/metadata.go

```diff
@@ -0,0 +1,17 @@
+package codec
+
+import (
+	"github.com/clinia/models-client-go/cliniamodel/common"
+	"go.jetify.com/ai/api"
+)
+
+// Metadata carries provider-specific per-call overrides extracted from ProviderMetadata.
+type Metadata struct {
+	// Requester allows callers/tests to inject a pre-constructed requester.
+	Requester common.Requester
```

Comment:
I feel like the reason of Metadata initially in the package is to hold metadata information sent during the call rather then holding the struct. Do we need to keep the requester here? I feel it could be created and dropped during every DoEmbed or DoX method call